src/decode.rs

```diff
@@ -3989,14 +3989,13 @@ unsafe fn setup_tile(
             &f.lf.lr_mask[sb_idx as usize].lr[p][unit_idx as usize]
         };
 
-        let lr = lr_ref.get();
+        let lr = lr_ref.try_write().unwrap();
         let lr = Av1RestorationUnit {
             filter_v: [3, -7, 15],
             filter_h: [3, -7, 15],
             sgr_weights: [-32, 31],
-            ..lr
+            ..*lr
         };
-        lr_ref.set(lr);
```

Comment:
Why are we no longer writing the value to `lr_ref`? It looks like the original was actually updating the data in `lr_ref`, but the new version doesn't.

Reply:
Good catch. It doesn't seem to affect our tests, but this is definitely wrong. Should be fixed now.